crates/polars-core/src/chunked_array/logical/datetime.rs

```diff
@@ -81,19 +81,16 @@ impl LogicalType for DatetimeChunked {
             },
             #[cfg(feature = "dtype-time")]
             (Datetime(tu, _), Time) => match tu {
-                TimeUnit::Nanoseconds => Ok((self.0.as_ref() % NS_IN_DAY)
-                    .cast(&Int64)
-                    .unwrap()
+                TimeUnit::Nanoseconds => Ok(self
+                    .apply_values(|v| v % NS_IN_DAY)
```

Comment:
this shortens the code - but also ends up hitting

https://github.com/pola-rs/polars/blob/2b43fc1ac1af84ed118ff3f8840d328a12c35510/crates/polars-core/src/chunked_array/ops/chunkops.rs#L102-L104